data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -102,58 +101,82 @@ export const setBatchUpsertRevisionMetadata = async (
             ]),
     );
 
-    // Get sourceId:sourceEntryId for cases that are included in the batch
-    // request but that won't be modified (because the request case is identical
-    // to the existing one).
-    const unchangedCaseIdSet = request.body.cases.reduce(
-        // TODO: Type request Cases.
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (unchangedIds: Set<string>, c: any) => {
-            if (c.caseReference?.sourceId && c.caseReference?.sourceEntryId) {
-                const existingCase = existingCasesByCaseRefCombo.get(
-                    c.caseReference.sourceId +
-                        ':' +
-                        c.caseReference.sourceEntryId,
-                );
-                if (existingCase !== undefined && existingCase.equalsJSON(c)) {
-                    unchangedIds.add(existingCase._id.toString());
-                }
+    const unchangedCaseIdSet = new Set();
+    request.body.cases = request.body.cases.filter((c: any) => {
```

Comment:
Could we filter this in-place (remove array items manually)? I almost never really worry about memory usage, but this is potentially double-digit MB, and we'd previously seen OOMs developing locally for 50k-100k cases. IIU[Javascript]C, `.filter` will always allocate a second array, so we'll have a peak of 2xNumCases in memory here.

Reply:
Done.